fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/ProjectOtherJoinConditionForNestedLoopJoin.java

```diff
@@ -0,0 +1,128 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite;
+
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.Alias;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.visitor.DefaultExpressionRewriter;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalProject;
+
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * join (l_orderkey > n_nationkey + n_regionkey)
+ *    +----scan(lineItem)
+ *    +----scan(nation)
+ * =>
+ * join(l_orderkey > x)
+ *    +----scan(lineItem)
+ *    +----project(n_nationkey + n_regionkey as x)
+ *         +----scan(nation)
+ */
+public class ProjectOtherJoinConditionForNestedLoopJoin extends OneRewriteRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalJoin().when(
+                join -> join.getHashJoinConjuncts().isEmpty()
+                        && !join.isMarkJoin()
+                        && !join.getOtherJoinConjuncts().isEmpty()
+                        && !join.getJoinReorderContext().isExtractSubExpressionForChild()
+        ).then(
+                join -> {
+                    join.getJoinReorderContext().setExtractSubExpressionForChild(true);
```

Comment:
why need this? just like push hash join conjuncts, when not changed, we return null then the rule will not fall into dead loop